### ISSUE TITLE
Add conformance test for GRPCRoute traffic splitting

### ISSUE THREAD

**Issue by pleshakov** [kubernetes-sigs/gateway-api#2901]
**What would you like to be added**:

A conformance test for traffic splitting in GRPCRoute -- https://github.com/kubernetes-sigs/gateway-api/blob/729d38f9bf17ff72d17f0148452416551296f67c/apis/v1alpha2/grpcroute_types.go#L271

**Why this is needed**:

Increase conformance feature coverage of GRPCRoute

**Comment by yashvardhan-kukreja** [comment]
Hi @pleshakov , I would love to contribute to this one.
/assign

**Comment by pleshakov** [comment]
Hi @yashvardhan-kukreja 

> Hi @pleshakov , I would love to contribute to this one.

that's great! I'd recommend to wait until any of the maintainers replied that this issue makes sense or ask this question in the gateway api slack channel https://gateway-api.sigs.k8s.io/contributing/#communications , to double check this work is needed.

**Comment by robscott** [comment]
I agree that this would be a useful test to have. @yashvardhan-kukreja or @pleshakov do either of you have time to work on this?

/remove-lifecycle stale

**Comment by ruixiansong** [comment]
I'm interested to work on this
